### PR TITLE
Backward functions and backprop support in operation functions

### PR DESCRIPTION
**Pull request by Jeli04**
Reorganized file structure to support additional function calls like ReLU and softmax 

Added operator overloading in Array class which is necessary for the matmul backward

Added more operator overloading in Tensor class

NOTE* backward function is unfinished

### CODE REVIEW

**Comment by nsnod** [comment]
looking good lil bro - good job asking your senior dev

**Comment by Jeli04** [comment]
Watch your tone buddy just remember who you're working for
